backend/test/modules/doubles-tournament/core/application/BookTableAndPublishEventAfterCallingMatch.spec.ts

```diff
@@ -0,0 +1,187 @@
+
+    //When
+    const callMatch = new CallMatch({
+      tournamentId: tournamentId,
+      matchFromQueue: { matchNumber: matchNumber, team1Id: team1Id, team2Id: team2Id },
+      tableNumber: tableNumber,
+    });
+    const commandResultFromCallingMatch = await doublesTournament.executeCommand(callMatch);
+
+    const bookTable = new BookTournamentTable(tournamentId, tableNumber);
+    const commandResultFromBookingTable = await doublesTournament.executeCommand(bookTable);
+
+    //Then
+    expect(commandResultFromCallingMatch.isSuccess()).toBeTruthy();
+    expect(commandResultFromBookingTable.isSuccess()).toBeTruthy();
+
+    expect(doublesTournament.lastPublishedEvent()).toStrictEqual(
+      new MatchWasCalled({
+        occurredAt: currentTime,
+        tournamentId: tournamentId,
+        matchFromQueue: { matchNumber: matchNumber, team1Id: team1Id, team2Id: team2Id },
+        tableNumber: tableNumber,
+      }),
+    );
+
+    expect(executeSpy).toHaveBeenCalledTimes(3);
+
+    const bookFromAvailableTables = new BookTournamentTable(tournamentId, tableNumber);
+    expect(commandBus.execute).toHaveBeenLastCalledWith(bookFromAvailableTables);
+  });
+
+  it('Then publish event and execute BookTournamentTable command from different module | table has been already booked', async () => {
+    //Given
+    const bookTournamentTableCommandHandler: CommandHandler<BookTournamentTable> = {
+      async execute(command: BookTournamentTable): Promise<CommandResult> {
+        return CommandResult.failureDueTo(new Error('Table has been already booked!'));
+      },
+    };
+
+    const commandBus = new InMemoryCommandBus();
+    commandBus.registerHandler(BookTournamentTable, bookTournamentTableCommandHandler);
+    const executeSpy = jest.spyOn(commandBus, 'execute');
+
+    const doublesTournament = testDoublesTournamentsModule(currentTime, entityIdGen, commandBus);
+
+    //When
+    const callMatch = new CallMatch({
+      tournamentId: tournamentId,
+      matchFromQueue: { matchNumber: matchNumber, team1Id: team1Id, team2Id: team2Id },
+      tableNumber: tableNumber,
+    });
+    const commandResultFromCallingMatch = await doublesTournament.executeCommand(callMatch);
+
+    const bookTable = new BookTournamentTable(tournamentId, tableNumber);
+    const commandResultFromBookingTable = await doublesTournament.executeCommand(bookTable);
+
+    //Then
+    expect(commandResultFromCallingMatch.isSuccess()).toBeTruthy();
+    expect((commandResultFromBookingTable as Failure).reason).toStrictEqual(new Error('Table has been already booked!'));
+
+    expect(doublesTournament.lastPublishedEvent()).toStrictEqual(
+      new MatchWasCalled({
+        occurredAt: currentTime,
+        tournamentId: tournamentId,
+        matchFromQueue: { matchNumber: matchNumber, team1Id: team1Id, team2Id: team2Id },
+        tableNumber: tableNumber,
+      }),
+    );
+
+    expect(executeSpy).toHaveBeenCalledTimes(3);
```

Comment:
A czemu nie 10? Co to sprawdza ;) ?

Reply:
no właśnie to mi średnio tu pasuje ( napisałem tez to wyżej w jednej odpowiedzi)